src/main/java/uk/gov/hmcts/ccd/domain/model/casedataaccesscontrol/matcher/BeginDateEndDateMatcher.java

```diff
@@ -26,4 +27,10 @@ public void matchAttribute(Pair<RoleAssignment, RoleMatchingResult> resultPair,
                 && roleAssignment.getEndTime().compareTo(now) > 0);
         }
     }
+
+    @Override
+    public void matchAttribute(Pair<RoleAssignment, RoleMatchingResult> resultPair,
+                               CaseTypeDefinition caseTypeDefinition) {
+
```

Comment:
Assuming this is empty as the check doesn't apply to the type? If so can we add a comment or something to make it clear why there's no implementation. Same for the other examples where we're doing this. Would expect Sonar to fail on this too.

Reply:
And maybe too late to be asking this question, but are there going to be any examples where we do actually have an implementation for both of the methods in `RoleAttributeMatcher`?

Reply:
instead of having different implementations for CaseDetails and CaseTypeDefinition, better to keep differnt methods in one interface, we have cases in JurisdictionMatcher and SecurityClassificationMatcher as well